src/Sparrow/Utils/NativeMemoryCleaner.cs

```diff
@@ -103,6 +103,10 @@ public void CleanNativeMemory(object state)
                     }
                 }
             }
+            catch (OutOfMemoryException)
```

Comment:
We should handle all errors here, no?

Reply:
Wasn't sure if we want to handle exceptions like double dispose or let it crash here intentionally.

Reply:
It isn't so much as that error, but needing to handle _any_ error here, because otherwise we kill the process.